examples/std_global_allocator.rs

```diff
@@ -7,14 +7,15 @@ use talc::*;
 static mut START_ARENA: [u8; 10000] = [0; 10000];
 
 #[global_allocator]
-// the mutex provided by the `spin` crate is used here as it's a sensible choice
+// The mutex provided by the `spin` crate is used here as it's a sensible choice
 static ALLOCATOR: Talck<spin::Mutex<()>, ClaimOnOom> =
-    // we need to use the ClaimOnOom OOM handler or similar as allocations may
-    // occur prior to invocation of the program entrypoint main(), so claiming some
-    // memory must be done on-demand
+    // An OOM handler such as ClaimOnOom is required, as allocations
+    // may occur prior to invoking the program entrypoint `main()`,
+    // thus needs to support claiming memory on demand
```

Comment:
```suggestion
    // thus on-demand support for claiming memory is needed
```

Reply:
I'd rather not use active voice here, but the original isn't great either. I'll rewrite it momentarily.

Reply:
[on-demand support] for XYZ != support for [XYZ on-demand]

But otherwise I also want to rewrite some of the rest of it (the active voice changes aren't ideal), but this is a step in the right direction, thanks. I'll suggest different changes.